gio/src/task.rs

```diff
@@ -60,11 +61,42 @@ impl Task {
         }
     }
 
+    #[doc(alias = "g_task_return_value")]
+    pub fn return_value<V: ValueType>(&self, result: &V) {
+        unsafe extern "C" fn value_free(value: *mut c_void) {
+            let _: glib::Value = from_glib_full(value as *mut glib::gobject_ffi::GValue);
+        }
+
+        unsafe {
+            ffi::g_task_return_pointer(
+                self.to_glib_none().0,
+                result.to_value().to_glib_full() as *mut _,
+                Some(value_free),
+            )
+        }
+    }
+
+    #[doc(alias = "g_task_propagate_value")]
+    pub fn propagate_value<V: ValueType>(&self) -> Result<V, glib::Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let value = ffi::g_task_propagate_pointer(self.to_glib_none().0, &mut error);
+            if !error.is_null() {
+                return Err(from_glib_full(error));
+            }
+            let value =
+                Option::<glib::Value>::from_glib_full(value as *mut glib::gobject_ffi::GValue)
+                    .unwrap_or_else(|| glib::Value::from_type(glib::types::Type::UNIT));
+            Ok(V::from_value(&value))
+        }
+    }
+
     #[doc(alias = "g_task_run_in_thread")]
-    pub fn run_in_thread<Q>(&self, task_func: Q)
+    pub fn run_in_thread<Q, V>(&self, task_func: Q)
     where
-        Q: FnOnce(&Self, Option<&glib::Object>, Option<&Cancellable>),
+        Q: FnOnce(&Self, Option<&glib::Object>, Option<&Cancellable>) -> Result<V, glib::Error>,
         Q: Send + 'static,
+        V: ValueType + ToSendValue,
```

Comment:
```suggestion
        V: ValueType + Send,
```

That's easier to read and probably also works :)